brion/detail/meshHDF5.h

```diff
@@ -75,174 +78,167 @@ class MeshHDF5 : public Mesh
+            {
+                SilenceHDF5 silence;
+                return HighFive::File(source,
+                                      overwrite ? H5F_ACC_TRUNC : H5F_ACC_EXCL);
+            }
+            catch (const HighFive::FileException& exc)
+            {
+                LBTHROW(std::runtime_error("Could not create mesh file " +
+                                           source + ": " + exc.what()));
+            }
+        }())
     {
-        try
-        {
-            SilenceHDF5 silence;
-            _file =
-                H5::H5File(source, overwrite ? H5F_ACC_TRUNC : H5F_ACC_EXCL);
-        }
-        catch (const H5::Exception& exc)
-        {
-            LBTHROW(std::runtime_error("Could not create mesh file " + source +
-                                       ": " + exc.getDetailMsg()));
-        }
     }
 
     virtual size_t getNumVertices() const
     {
-        const H5::DataSet& dataset =
-            _file.openDataSet("/membrane/mesh/vertices");
-        return _numElements(dataset);
+        return _numElements(_file.getDataSet("/membrane/mesh/vertices"));
     }
 
     virtual Vector3fsPtr readVertices() const
     {
-        const H5::DataSet& dataset =
-            _file.openDataSet("/membrane/mesh/vertices");
+        const auto dataset = _file.getDataSet("/membrane/mesh/vertices");
         Vector3fsPtr buffer(new Vector3fs);
         buffer->resize(_numElements(dataset));
-        dataset.read(buffer->data(), H5::PredType::IEEE_F32LE);
+        dataset.read(*buffer);
         return buffer;
     }
 
     virtual uint16_tsPtr readVertexSections() const
     {
-        const H5::DataSet& dataset =
-            _file.openDataSet("/membrane/mesh/mappings/vertex/section_ids");
+        const auto dataset =
+            _file.getDataSet("/membrane/mesh/mappings/vertex/section_ids");
         uint16_tsPtr buffer(new uint16_ts);
         buffer->resize(_numElements(dataset));
-        dataset.read(buffer->data(), H5::PredType::STD_U32LE);
+        dataset.read(*buffer);
         return buffer;
     }
 
     virtual floatsPtr readVertexDistances() const
     {
-        const H5::DataSet& dataset = _file.openDataSet(
+        const auto dataset = _file.getDataSet(
             "/membrane/mesh/mappings/vertex/relative_positions");
         floatsPtr buffer(new floats);
         buffer->resize(_numElements(dataset));
-        dataset.read(buffer->data(), H5::PredType::IEEE_F32LE);
+        dataset.read(*buffer);
         return buffer;
     }
 
     virtual size_t getNumTriangles() const
     {
-        const H5::DataSet& dataset =
-            _file.openDataSet("/membrane/mesh/triangles/soup");
-        return _numElements(dataset);
+        return _numElements(_file.getDataSet("/membrane/mesh/triangles/soup"));
     }
 
     virtual uint32_tsPtr readTriangles() const
     {
-        const H5::DataSet& dataset =
-            _file.openDataSet("/membrane/mesh/triangles/soup");
+        return uint32_tsPtr();
```

Comment:
rm

Reply:
I could very well remove the whole reader. The fact that there are are no tests seems relevant.
I think it's not used anywhere.